DeUrgenta/Features/Authentication/SignIn/Coordinator/SignInCoordinator.swift

```diff
@@ -6,22 +6,37 @@ final class SignInCoordinator: Coordinator {
     var navigationController: UINavigationController
     var childCoordinators: [Coordinator] = []
 
+    var signInViewModel = SignInViewModel()
+
     init(navigationController: UINavigationController) {
         self.navigationController = navigationController
     }
 
     func start() {
-        let viewController = UIHostingController(rootView: SignInView(delegate: self))
+        let viewController = UIHostingController(rootView: SignInView(viewModel: signInViewModel, delegate: self))
         navigationController.pushViewController(viewController, animated: true)
     }
 }
 
-extension SignInCoordinator: SignInViewDelegate {
+extension SignInCoordinator: SignInViewDelegate, ResetPasswordViewDelegate, InsertNewPasswordViewDelegate {
     func signInViewDidTapSignIn(_: SignInView) {
         // TODO: Implement this action
     }
 
     func signInViewDidTapForgotPassword(_: SignInView) {
-        // TODO: Implement this action
+        let viewController = UIHostingController(rootView: ResetPasswordView(delegate: self))
+        navigationController.pushViewController(viewController, animated: true)
+    }
+
+    func resetPasswordViewDidTapReset(_: ResetPasswordView, for _: String) {
+        let viewController = UIHostingController(rootView: InsertNewPasswordView(delegate: self))
+        navigationController.pushViewController(viewController, animated: true)
+    }
+
+    func insertNewPasswordViewDidTapSave(_: InsertNewPasswordView) {
+        let viewControllers: [UIViewController] = navigationController.viewControllers as [UIViewController]
+        navigationController.popToViewController(viewControllers[viewControllers.count - 3], animated: true)
```

Comment:
Can we find a way to pop to a view controller without using the -3 index? This can be tricky in the feature. Maybe filter existing controller with as? <type of the view controller you are looking for>?

Reply:
ok